lib/tasks/whitehall_importer.rb

```diff
@@ -120,37 +127,93 @@ def create_edition(document, translation, whitehall_edition, edition_number)
         created_at: whitehall_edition["created_at"],
       )
 
-      Edition.create!(
+      edition = Edition.create!(
         document: document,
         number: edition_number,
         revision_synced: true,
         revision: revision,
-        status: Status.new(
-          state: state(whitehall_edition),
-          revision_at_creation: revision,
-        ),
+        status: initial_status(whitehall_edition, revision),
         current: whitehall_edition["id"] == most_recent_edition["id"],
         live: live?(whitehall_edition),
         created_at: whitehall_edition["created_at"],
         updated_at: whitehall_edition["updated_at"],
         created_by_id: user_ids[first_author["whodunnit"]],
         last_edited_by_id: user_ids[last_author["whodunnit"]],
       )
+
+      set_current_status(whitehall_edition, edition) if whitehall_edition["state"] == "withdrawn"
+    end
+
+    def initial_status(whitehall_edition, revision)
+      author = if whitehall_edition["state"] == "withdrawn"
+                 published_author(whitehall_edition)
+               else
+                 current_author(whitehall_edition)
+               end
```

Comment:
I feel like author might be a bit misleading as a variable name here as you'd expect that to be a user object whereas this returns a history entry.

Reply:
I copied the name from here: https://github.com/alphagov/content-publisher/blob/master/lib/tasks/whitehall_importer.rb#L87 but I'm happy to change it.

Reply:
I've altered this to revision_history_event, and the author method name's to have a suffix of `_history_event` which makes the intent of these methods a bit clearer.